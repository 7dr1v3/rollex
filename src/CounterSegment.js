import React from 'react'
const { arrayOf, string } = React.PropTypes

class CounterSegment extends React.Component {
  static propTypes = {
    digits: arrayOf(string).isRequired,
    label: string.isRequired
  }

  render () {
    const digits = this.props.digits.map((digit, index) => {
      return (
        <div className='cntr-digit' key={index}>
          { digit }
        </div>
      )
    })
    return (
      <div className='cntr-segment'>
        { digits }
        <div className='cntr-label'>
          { this.props.label }
        </div>
      </div>
    )
  }
}

export default CounterSegment
