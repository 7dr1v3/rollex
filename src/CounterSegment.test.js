import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import CounterSegment from './CounterSegment'

it('matches snapshot', function () {
  const component = shallow(<CounterSegment label='days' digits={['0', '0']} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

it('should render a .cntr-digit for each digit', function () {
  const component = shallow(<CounterSegment label='days' digits={['0', '1', '2']} />)
  expect(component.find('.cntr-digit').length).toEqual(3)
})
