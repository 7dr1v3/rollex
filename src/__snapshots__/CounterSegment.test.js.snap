// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="cntr-segment"
>
  <div
    className="cntr-digit"
  >
    0
  </div>
  <div
    className="cntr-digit"
  >
    0
  </div>
  <div
    className="cntr-label"
  >
    days
  </div>
</div>
`;
